File: img/logos/generate_assets.mjs
/**
 * Generate PNG/ICO assets from SVG logos for AboRadar.
 * Uses sharp (Node.js) - works on Windows without Cairo.
 */
import sharp from 'sharp';
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);
const ABORADAR_DIR = path.resolve(__dirname, '../..');
const APP_DIR = path.resolve(ABORADAR_DIR, '../aboradar-app');

const ICON_SVG = path.join(__dirname, 'logo-icon.svg');

async function svgToPng(svgPath, outputPath, width, height) {
  height = height || width;
  const svgBuffer = fs.readFileSync(svgPath);
  await sharp(svgBuffer)
    .resize(width, height)
    .png()
    .toFile(outputPath);
  const size = (fs.statSync(outputPath).size / 1024).toFixed(1);
  console.log(`  Created: ${path.relative(ABORADAR_DIR, outputPath)} (${width}x${height}, ${size}KB)`);
}

async function createOgImage(svgPath, outputPath) {
  // Dark background 1200x630
  const bg = sharp({
    create: {
      width: 1200,
      height: 630,
      channels: 4,
      background: { r: 8, g: 9, b: 13, alpha: 1 }
    }
  }).png();

  // Render logo
  const logoSize = 280;
  const svgBuffer = fs.readFileSync(svgPath);
  const logoPng = await sharp(svgBuffer)
    .resize(logoSize, logoSize)
    .png()
    .toBuffer();

  // Composite logo centered
  await bg
    .composite([{
      input: logoPng,
      left: Math.round((1200 - logoSize) / 2),
      top: Math.round((630 - logoSize) / 2) - 30,
    }])
    .toFile(outputPath);

  const size = (fs.statSync(outputPath).size / 1024).toFixed(1);
  console.log(`  Created: ${path.relative(ABORADAR_DIR, outputPath)} (1200x630, ${size}KB)`);
}

async function createFaviconIco(pngPath, icoPath) {
  // Create a simple ICO by using 32x32 PNG wrapped in ICO format
  const png32 = await sharp(pngPath).resize(32, 32).png().toBuffer();
  const png16 = await sharp(pngPath).resize(16, 16).png().toBuffer();

  // ICO file format: header + directory entries + image data
  const images = [png16, png32];
  const sizes = [16, 32];

  // ICO Header (6 bytes)
  const header = Buffer.alloc(6);
  header.writeUInt16LE(0, 0); // Reserved
  header.writeUInt16LE(1, 2); // Type: ICO
  header.writeUInt16LE(images.length, 4); // Number of images

  // Directory entries (16 bytes each)
  let offset = 6 + images.length * 16;
  const entries = [];
  for (let i = 0; i < images.length; i++) {
    const entry = Buffer.alloc(16);
    entry.writeUInt8(sizes[i] === 256 ? 0 : sizes[i], 0); // Width
    entry.writeUInt8(sizes[i] === 256 ? 0 : sizes[i], 1); // Height
    entry.writeUInt8(0, 2);  // Color palette
    entry.writeUInt8(0, 3);  // Reserved
    entry.writeUInt16LE(1, 4);  // Color planes
    entry.writeUInt16LE(32, 6); // Bits per pixel
    entry.writeUInt32LE(images[i].length, 8);  // Image size
    entry.writeUInt32LE(offset, 12); // Offset
    entries.push(entry);
    offset += images[i].length;
  }

  const ico = Buffer.concat([header, ...entries, ...images]);
  fs.writeFileSync(icoPath, ico);
  const size = (fs.statSync(icoPath).size / 1024).toFixed(1);
  console.log(`  Created: ${path.relative(ABORADAR_DIR, icoPath)} (ICO multi-size, ${size}KB)`);
}

async function main() {
  console.log('=== AboRadar Logo Asset Generator ===\n');

  // Ensure directories exist
  fs.mkdirSync(path.join(ABORADAR_DIR, 'img'), { recursive: true });
  fs.mkdirSync(path.join(APP_DIR, 'assets'), { recursive: true });

  // 1. App Icon (1024x1024)
  console.log('1. App Icon (1024x1024):');
  await svgToPng(ICON_SVG, path.join(APP_DIR, 'assets', 'icon.png'), 1024);

  // 2. Favicon
  console.log('\n2. Favicon:');
  const favicon512 = path.join(__dirname, 'favicon-512.png');
  await svgToPng(ICON_SVG, favicon512, 512);
  await createFaviconIco(favicon512, path.join(ABORADAR_DIR, 'img', 'favicon.ico'));
  await svgToPng(ICON_SVG, path.join(ABORADAR_DIR, 'img', 'favicon.png'), 32);

  // 3. Web sizes
  console.log('\n3. Web sizes:');
  for (const size of [192, 180, 128]) {
    await svgToPng(ICON_SVG, path.join(__dirname, `logo-icon-${size}.png`), size);
  }

  // 4. OG Image
  console.log('\n4. OG Image (1200x630):');
  await createOgImage(ICON_SVG, path.join(ABORADAR_DIR, 'img', 'og-image.png'));

  // 5. Splash Icon
  console.log('\n5. Splash Icon:');
  await svgToPng(ICON_SVG, path.join(APP_DIR, 'assets', 'splash-icon.png'), 512);

  // 6. Android icons
  console.log('\n6. Android Adaptive Icons:');
  await svgToPng(ICON_SVG, path.join(APP_DIR, 'assets', 'android-icon-foreground.png'), 1024);
  await svgToPng(ICON_SVG, path.join(APP_DIR, 'assets', 'android-icon-monochrome.png'), 1024);

  // 7. Expo web favicon
  console.log('\n7. Expo Web Favicon:');
  await svgToPng(ICON_SVG, path.join(APP_DIR, 'assets', 'favicon.png'), 48);

  console.log('\n=== Done! All assets generated. ===');
}

main().catch(console.error);

File: img/logos/generate_assets.py
"""Generate PNG assets from SVG logo files for AboRadar."""
import os
import sys

try:
    import cairosvg
except ImportError:
    print("cairosvg not available, trying alternative...")
    sys.exit(1)

from PIL import Image
import io

BASE_DIR = os.path.dirname(os.path.abspath(__file__))
ABORADAR_DIR = os.path.dirname(os.path.dirname(BASE_DIR))
APP_DIR = os.path.join(os.path.dirname(ABORADAR_DIR), "aboradar-app")

ICON_SVG = os.path.join(BASE_DIR, "logo-icon.svg")
FULL_SVG = os.path.join(BASE_DIR, "logo-full.svg")


def svg_to_png(svg_path, output_path, width, height=None):
    """Convert SVG to PNG at specified dimensions."""
    if height is None:
        height = width
    png_data = cairosvg.svg2png(
        url=svg_path,
        output_width=width,
        output_height=height,
    )
    with open(output_path, "wb") as f:
        f.write(png_data)
    size_kb = os.path.getsize(output_path) / 1024
    print(f"  Created: {output_path} ({width}x{height}, {size_kb:.1f}KB)")


def create_favicon_ico(png_path, ico_path):
    """Create .ico from PNG with multiple sizes."""
    img = Image.open(png_path)
    sizes = [(16, 16), (32, 32), (48, 48)]
    imgs = [img.resize(s, Image.LANCZOS) for s in sizes]
    imgs[0].save(ico_path, format="ICO", sizes=sizes, append_images=imgs[1:])
    size_kb = os.path.getsize(ico_path) / 1024
    print(f"  Created: {ico_path} (multi-size ICO, {size_kb:.1f}KB)")


def create_og_image(svg_path, output_path):
    """Create 1200x630 OG image with logo centered on dark background."""
    # Create dark background
    bg = Image.new("RGBA", (1200, 630), (8, 9, 13, 255))

    # Render logo at reasonable size
    logo_size = 280
    png_data = cairosvg.svg2png(
        url=svg_path,
        output_width=logo_size,
        output_height=logo_size,
    )
    logo = Image.open(io.BytesIO(png_data)).convert("RGBA")

    # Center the logo (slightly left to leave room for text)
    x = (1200 - logo_size) // 2
    y = (630 - logo_size) // 2 - 40
    bg.paste(logo, (x, y), logo)

    # Add a subtle text area below (brand name as simple bar)
    # We'll keep it simple - just the icon centered
    # The wordmark would need font rendering which is complex

    bg.save(output_path, "PNG")
    size_kb = os.path.getsize(output_path) / 1024
    print(f"  Created: {output_path} (1200x630, {size_kb:.1f}KB)")


def create_splash_icon(svg_path, output_path):
    """Create splash screen icon (centered on dark bg)."""
    # Splash icon for Expo: just the logo on transparent bg
    size = 512
    png_data = cairosvg.svg2png(
        url=svg_path,
        output_width=size,
        output_height=size,
    )
    with open(output_path, "wb") as f:
        f.write(png_data)
    size_kb = os.path.getsize(output_path) / 1024
    print(f"  Created: {output_path} ({size}x{size}, {size_kb:.1f}KB)")


def main():
    print("=== AboRadar Logo Asset Generator ===\n")

    # 1. App Icon (1024x1024) for Expo
    print("1. App Icon (1024x1024):")
    app_icon_path = os.path.join(APP_DIR, "assets", "icon.png")
    svg_to_png(ICON_SVG, app_icon_path, 1024)

    # 2. Favicon PNGs
    print("\n2. Favicon:")
    favicon_png = os.path.join(BASE_DIR, "favicon-512.png")
    svg_to_png(ICON_SVG, favicon_png, 512)

    favicon_32 = os.path.join(BASE_DIR, "favicon-32.png")
    svg_to_png(ICON_SVG, favicon_32, 32)

    # Create ICO
    ico_path = os.path.join(ABORADAR_DIR, "img", "favicon.ico")
    create_favicon_ico(favicon_png, ico_path)

    # Also save a 32px PNG favicon
    favicon_png_web = os.path.join(ABORADAR_DIR, "img", "favicon.png")
    svg_to_png(ICON_SVG, favicon_png_web, 32)

    # 3. Various sizes for web
    print("\n3. Web sizes:")
    for size in [192, 180, 128]:
        out = os.path.join(BASE_DIR, f"logo-icon-{size}.png")
        svg_to_png(ICON_SVG, out, size)

    # 4. OG Image
    print("\n4. OG Image (1200x630):")
    og_path = os.path.join(ABORADAR_DIR, "img", "og-image.png")
    create_og_image(ICON_SVG, og_path)

    # 5. Splash Icon for Expo
    print("\n5. Splash Icon:")
    splash_path = os.path.join(APP_DIR, "assets", "splash-icon.png")
    create_splash_icon(ICON_SVG, splash_path)

    # 6. Android adaptive icon (foreground)
    print("\n6. Android Adaptive Icon:")
    android_fg = os.path.join(APP_DIR, "assets", "android-icon-foreground.png")
    svg_to_png(ICON_SVG, android_fg, 1024)

    # Monochrome version (same icon)
    android_mono = os.path.join(APP_DIR, "assets", "android-icon-monochrome.png")
    svg_to_png(ICON_SVG, android_mono, 1024)

    # Expo web favicon
    print("\n7. Expo Web Favicon:")
    expo_favicon = os.path.join(APP_DIR, "assets", "favicon.png")
    svg_to_png(ICON_SVG, expo_favicon, 48)

    print("\n=== Done! All assets generated. ===")


if __name__ == "__main__":
    main()
